Bonsai.Core/WorkflowBuildException.cs

```diff
@@ -88,6 +88,9 @@ public WorkflowBuildException(string message, ExpressionBuilder builder, Excepti
         /// The <see cref="StreamingContext"/> that contains contextual information about
         /// the source or destination.
         /// </param>
+#if NET8_0_OR_GREATER
+        [Obsolete(DiagnosticId = "SYSLIB0051")]
+#endif
         protected WorkflowBuildException(SerializationInfo info, StreamingContext context)
```

Comment:
I don't think they are used, the only reason I left them for now is out of worry that someone might have inherited from these classes and implemented their constructors. I have seen at least some scripts and third-party packages try to at least throw using these classes. In case they would have inherited from them and implemented these constructors, I think the type loader might crash when loading the package.

Or maybe you were asking why not remove them completely from .NET 8.0 onwards? If that was the suggestion I agree we probably should do that since there is no one using the .NET core assemblies at scale.

Reply:
I was asking a little bit of both. Limiting the removal to .NET 8 seems good and fine to me, it's an easy fix for people to just remove their own overload when they migrate if they happen to be extending these classes.